arch/rv32i/src/epmp.rs

```diff
@@ -318,7 +318,7 @@ impl PMPRegion {
             (region_start, region_end)
         };
 
-        if region_start < other_end && other_start < region_end {
+        if region_start < (other_end - 4) && other_start < (region_end - 4) {
```

Comment:
Actually, can we add a comment here stating what is in this PR description? Otherwise at a glance this just looks like a bug